### DISCUSSION TITLE
Purpose of the constants

### DISCUSSION THREAD

**Discussion by pendraggon87** [Dotnet-Boxed/Templates#36] [Q&A]
Is there a purpose behind using the constants? It seems that this takes more time in development, and precludes being able to use ROUTE("") for the Index action of a controller?

**Comment by RehanSaeed** [comment]
There are two reasons.
1. We use named routes instead of controller name and action name. This has better performance. So `Url.RouteUrl("RouteName")` is quicker than `Url.Action("Controller", "Action")`.
2. What happens when you've finished your application, you are using strings everywhere and now you want to rename a controller or action? You have to go through all the views looking for any string that has the old controller or action name. Having constants allows Visual Studio to do all the work and lets us rename the constant easily.

Yes it takes longer but in the long run you save time with number two. At the end of the day, it's up to you so you can strip them out if you don't like them.
